Justfile

```diff
@@ -174,3 +174,16 @@ check-protocol-schema:
 
 check_build_public_libraries:
     cargo check {{public_libraries}}
+
+# Ensures all published crates can be built in isolation.
+# Generally CI tests combinations of crates, which may miss a scenario where a crate cannot be
+# built in isolation. For example because a crucial feature is enabled only accidentally by
+# another crate included in the build process.
+#
+# By default `cargo workspaces list` ignores private crates.
+check_published_crates_in_isolation:
+    #!/usr/bin/env bash
+    set -euxo pipefail
+    for crate in `cargo workspaces list`; do
+        cargo check -p $crate;
```

Comment:
Nit: you will need to `cargo publish --dry-run`. One of the problems we had was precisely about packaging particularities. Besides that we most likely want to use `cargo-ws run` and/or have it version publishable crates beforehand too… though IDK how well that'll work for dependencies with publishing being `--dry-run`...

Reply:
> we most likely want to use `cargo-ws run`

Could you mean [cargo workspaces exec](https://github.com/pksunkara/cargo-workspaces?tab=readme-ov-file#exec)? I know `cargo ws` is a shorthand for `cargo workspaces` but there's also a separate [cargo-ws](https://crates.io/crates/cargo-ws) crate.

Another option might be [[ref](https://github.com/pksunkara/cargo-workspaces?tab=readme-ov-file#publish)]:

```sh
cargo workspaces publish --dry-run --from-git
```

Reply:
> have it version publishable crates beforehand too

There's also [`cargo workspaces version`](https://github.com/pksunkara/cargo-workspaces?tab=readme-ov-file#version), though I'm not sure if/how this would work in CI as it commits changes to git.

Reply:
re `cargo workspaces`: use exactly the same helper tool that we use for publishing. It comes from our fork of `cargo-workspaces`. See the implementation of the publishing workflows for exact details.

Reply:
Notably it has a pretty significant divergence in features between itself and upstream, so I wouldn't rely on the upstream documentation too much.

Reply:
Well, the goal is to prevent situation where we have a release branch, and during the last 5 minutes of the release rites it turns out that everything is broken beyond repair all because somebody's no-op refactor broke things in a way that's super hard to go revert because its a small part of somebody's huge feature PR. I guess another way to look at this: our deliverable is not just a build of `neard` that passes the tests. We *also* deliver a collection of packages for contract developers to use. And the hope is to see this process to not break.

I also want to push back against a partial solution here, precisely because this leads to accumulation of a bunch of partial solutions (see e.g. the Justfile task right above the one you've just added) but none of them provide full guarantees that the release will pass without exploding on us.